src/Search.php

```diff
@@ -6557,8 +6557,20 @@ public static function giveItem(
             $linkfield = $so["linkfield"];
 
            /// TODO try to clean all specific cases using SpecificToDisplay
-
             switch ($table . '.' . $field) {
+                case "glpi_locations.completename":
+                case "glpi_groups.completename":
+                case "glpi_itilcategories.completename":
+                    if (
+                        $itemtype == Ticket::class // only for helpdesk
+                        && $data[$ID][0]['name'] != null //column have value in DB
+                        && $_SESSION['glpiuse_flat_dropdowntree_on_search_result'] //user want the short name
+                    ) {
```

Comment:
Why does this preference is not applied on all search results related to a `completename` field?